service/storage_service.cc

```diff
@@ -154,6 +154,14 @@ namespace {
 
 static constexpr std::chrono::seconds wait_for_live_nodes_timeout{30};
 
+static future<> run_on_repair_service(sharded<repair_service>& repair, std::function<future<>(repair_service&)> f) {
+    if (!repair.local_is_initialized()) {
+        throw seastar::abort_requested_exception();
+    }
+    gate::holder holder = repair.local().async_gate().hold();
```

Comment:
This seems so simple to do for callers, I think they can just do this instead of having this method, no?

Reply:
If it's only the gate then sure, but I think we should check `local_is_initialized` before and it will make it kind of messy, no?

Reply:
Yes, indeed.